colorbleed/scripts/fusion/deadline_switch_and_submit.py

```diff
@@ -0,0 +1,221 @@
+
+    See `get_fusion_instance` function for more details
+
+In `avalon.fusion.pipeline` we have create a work around to get the fusion
+instance. This is done through
+
+    getattr(sys.module["__main__"], "fusion", None)
+
+Because we do this we can also allow to set correct fusion module, this is done
+by using the setattr. This will ensure that all other functions which are run
+within `process()` can find `fusion`.
+
+"""
+
+import subprocess
+import traceback
+import logging
+import time
+import sys
+import os
+
+log = logging.getLogger(__name__)
+
+# This script only works with Python 2.7 and 3.6
+version = "{0}{1}".format(*sys.version_info)  # {major}{minor}
+assert version in ["27", "36"], "Script only works in Python 2.7 or 3.6"
+key = "FUSION_PYTHON{0}_HOME".format(version)
+
+# Importing BlackmagicFusion package in standalone Python interpreter
+# crashes when not installed on default location but runs from, e.g. a
+# network share. Forcing Fusion's Python home magically fixes it.
+print("Setting %s to Python executable directory.." % key)
+os.environ[key] = os.path.dirname(sys.executable)
+
+# TODO: define these paths somewhere else
+FUSCRIPT_EXE = r"C:/Program Files/Blackmagic Design/Fusion9/FuScript.exe"
+FUSION_CONSOLE_EXE = r"C:/Program Files/Blackmagic Design/Fusion Render Node 9/FusionConsoleNode.exe"
+
+# Pipeline and config imports
+import avalon.fusion
+from avalon import api
+import colorbleed.fusion.lib as fusionlib
+
+# Application related imports
+import BlackmagicFusion as bmf
+
+
+def start_server():
+    bmf.startserver()
+    return get_server()
+
+
+def get_server(tries=10, timeout=0.5):
+
+    count = 0
+    srv = None
+
+    while not srv:
+        count += 1
+        print("Connecting to ScriptServer (try: %s)" % count)
+        srv = bmf.scriptapp("", "localhost", timeout)  # Runs script server
+        if count > tries:
+            break
+
+    return srv
+
+
+def get_fusion_instance(pid, srv, timeout=10):
+    """Get the fusion instance which has been launched"""
+
+    count = 0
+    host = None
+    while not host:
+        if count > timeout:
+            break
+        fusion_hosts = srv.GetHostList().values()
+        host = next((i for i in fusion_hosts if int(i["ProcessID"]) == pid),
+                    None)
+        if not host:
+            print("Find Fusion host... (%ss)" % count)
+            time.sleep(0.5)
+            count += 0.5
+
+    assert host, "Fusion not found with pid: %s" % pid
+
+    return bmf.scriptapp(host["Name"], "localhost", 2, host["UUID"])
+
+
+def submit(current_comp):
+    """Set rendermode to deadline and publish / submit comp"""
+
+    # Set comp render mode to deadline
+    current_comp.SetData("colorbleed.rendermode", "deadline")
+
+    error_format = "Failed {plugin.__name__}: {error} -- {error.traceback}"
+
+    # Publish
+    context = api.publish()
+    if not context:
+        log.warning("Nothing collected.")
+        sys.exit(1)
```

Comment:
Shouldn't this also raise an error instead of hard exiting? Maybe the logged message could just be the raised message.

```python
raise RuntimeError("Nothing collected for publish")
```